src/components/charlist/RollDiceButton.tsx

```diff
@@ -0,0 +1,162 @@
+import CloseIcon from '@mui/icons-material/Close';
+import {
+  Box,
+  Button,
+  Menu,
+  MenuItem,
+  Tooltip,
+  Typography,
+  styled,
+} from '@mui/material';
+import { grey } from '@mui/material/colors';
+import D4Icon from 'assets/images/dice/D4.svg';
+import D6Icon from 'assets/images/dice/D6.svg';
+import D8Icon from 'assets/images/dice/D8.svg';
+import D10Icon from 'assets/images/dice/D10.svg';
+import D12Icon from 'assets/images/dice/D12.svg';
+import D20Icon from 'assets/images/dice/D20.svg';
+import D100Icon from 'assets/images/dice/D100.svg';
+import { FC } from 'react';
+import React from 'react';
+
+import ParBox from '../styled/ParBox';
+import { DiceMenu } from './interfaces';
+
+const StyledMenuItem = styled(MenuItem)(({ theme }) => ({
+  borderRadius: '50%',
+  backgroundColor: theme.palette.primary.main,
+  boxShadow: '5',
+  marginBottom: '0.5rem',
+  padding: '0.5rem',
+  '&:hover': {
+    backgroundColor: theme.palette.primary.light,
+  },
+}));
+
+const StyledMenu = styled(Menu)(({ theme }) => ({
+  '& .MuiPaper-root': {
+    backgroundColor: 'rgba(0, 0, 0, 0)',
+    boxShadow: 'none',
+  },
+}));
+
+const RollDiceButton: FC = () => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const dice: Array<DiceMenu> = [
+    {
+      name: 'D20',
+      iconName: D20Icon,
+      onClick: () => {
```

Comment:
You can write this onClick in the map and put all the data about the dices in another file